src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -9262,6 +9327,115 @@ void SketchObject::setExpression(const App::ObjectIdentifier& path,
     }
 }
 
+Part::TopoShape SketchObject::getEdge(const Part::Geometry *geo, const char *name) const
+{
+    Part::TopoShape shape(geo->toShape());
+    // shape.setElementName(Data::IndexedName::fromConst("Edge", 1),
+    //                      Data::MappedName::fromRawData(name));
+    TopTools_IndexedMapOfShape vmap;
+    TopExp::MapShapes(shape.getShape(), TopAbs_VERTEX, vmap);
+    std::ostringstream ss;
+    for(int i=1;i<=vmap.Extent();++i) {
+        auto gpt = BRep_Tool::Pnt(TopoDS::Vertex(vmap(i)));
+        Base::Vector3d pt(gpt.X(),gpt.Y(),gpt.Z());
+        PointPos pos[] = {PointPos::start,PointPos::end};
+        for(size_t j=0;j<sizeof(pos)/sizeof(pos[0]);++j) {
+            if(getPoint(geo,pos[j]) == pt) {
+                ss.str("");
+                ss << name << 'v' << static_cast<int>(pos[j]);
+                // shape.setElementName(Data::IndexedName::fromConst("Vertex", i),
+                //                      Data::MappedName::fromRawData(ss.str().c_str()));
+                break;
+            }
+        }
+    }
+    return shape;
+}
+
+Data::IndexedName SketchObject::checkSubName(const char *sub) const
+{
+    // FIXME: trivial implementation needs to be replaced with full logic
+    (void)sub;
+    return Data::IndexedName();
+}
+
+bool SketchObject::geoIdFromShapeType(const Data::IndexedName & indexedName,
+                                      int &geoId,
+                                      PointPos &posId) const
+{
+    posId = PointPos::none;
+    geoId = Sketcher::GeoEnum::GeoUndef;
+    if (!indexedName)
+        return false;
+    const char *shapetype = indexedName.getType();
+    if (boost::equals(shapetype,"Edge") ||
+        boost::equals(shapetype,"edge")) {
+        geoId = indexedName.getIndex() - 1;
+    } else if (boost::equals(shapetype,"ExternalEdge")) {
+        geoId = indexedName.getIndex() - 1;
+        geoId = Sketcher::GeoEnum::RefExt - geoId;
+    } else if (boost::equals(shapetype,"Vertex") ||
+               boost::equals(shapetype,"vertex")) {
+        int VtId = indexedName.getIndex() - 1;
+        getGeoVertexIndex(VtId,geoId,posId);
+        if (posId==PointPos::none) return false;
+    } else if (boost::equals(shapetype,"H_Axis")) {
+        geoId = Sketcher::GeoEnum::HAxis;
+    } else if (boost::equals(shapetype,"V_Axis")) {
+        geoId = Sketcher::GeoEnum::VAxis;
+    } else if (boost::equals(shapetype,"RootPoint")) {
+        geoId = Sketcher::GeoEnum::RtPnt;
+        posId = PointPos::start;
+    } else
+        return false;
+    return true;

```

Comment:
It might be a backwards-compatibility thing for files from older versions of @realthunder's fork. I am OK with removing it unless @JohnAD (or @realthunder of course) has an alternate opinion on the matter.

Reply:
Without knowing the original intent, I'd normally be reluctant to remove it. It's new code to this project, but their might be some odd border case discovered in the fork that requires it. @realthunder do you recalll why this was added in?

Reply:
Let's leave it alone, absent further input.

Reply:
The case variation is intentional. Because `Edge1` in sketch may refer to different edges depending on whether it is in edit mode, I use lower case `edge1` to refer to the edge in edit mode, so that `Sketch.getSubObject('edge1')` can return the correct edge.

As to why the code you referenced treats both `Edge` and `edge` the same is kind of complicated. In short, it is indeed to handle corner cases where other part of the code in FreeCAD outside the core that still use the old element name to reference sketch geometry while editing.